packages/core/src/connections/Connection.ts

```diff
@@ -84,6 +84,13 @@ export abstract class Connection {
   }
 
   protected logQuery(query: string, took?: number, language?: string): void {
+    const logger = this.config.getLogger();
+
+    // We only actually log something when debugMode is enabled. If it's not enabled,
+    // we can jump out here as a performance optimization and save ourselves some cycles
+    // preparing a message that won't get used.
+    if (!logger.debugMode) return;
```

Comment:
mmm how come this is not checked via eslint :]

```suggestion
    if (!logger.debugMode) {
      return;
    }
```